lib/vector-core/src/stream/batcher/reducer.rs

```diff
@@ -0,0 +1,44 @@
+use super::data::BatchData;
+
+/// A batch reducer based on a function.
+pub struct FunctionReducer<F, S> {
+    reducer: F,
+    state: S,
+    len: usize,
+}
+
+impl<F, S> FunctionReducer<F, S>
+where
+    S: Default,
+{
+    /// Creates a `FunctionReducer` with the given reducer function and default state.
+    pub fn with_default_state(reducer: F) -> Self {
```

Comment:
TIOLI _bikeshedding_ I would just call this `fn from_reducer`.

Reply:
Oh, doh... I did not mean to leave these changes in. I was originally thinking of trying to do this collapsing in the batcher itself, and then realize the existing code didn't necessarily provide enough of the extensibility that I needed to do so, and went with the more brute force approach.

I'll rewind the changes in this area.

Reply:
Backed out the changes in this area in 9179db442.